### PR TITLE
Add an extension point on the UserProviderWithPayloadSupportInterface

### PR DESCRIPTION
**Pull request by sroze**
So anybody can plug their own user provider and get access to the payload :)

### CODE REVIEW

**Comment by chalasr** [comment]
Great! One thing: PHP5 is still supported for this major :)

File: Security/Guard/JWTTokenAuthenticator.php [hunk 1]
```diff
@@ -17,6 +17,7 @@
 use Lexik\Bundle\JWTAuthenticationBundle\Security\Authentication\Token\JWTUserToken;
 use Lexik\Bundle\JWTAuthenticationBundle\Security\Authentication\Token\PreAuthenticationJWTUserToken;
 use Lexik\Bundle\JWTAuthenticationBundle\Security\User\JWTUserProvider;
```
Comment (chalasr) [inline Security/Guard/JWTTokenAuthenticator.php]: This use is now unused

**Comment by chalasr** [comment]
Thank you @sroze

**Comment by srigi** [comment]
This is merged, but not released as a tagged composer package.
Please release this, our team would much appreciated that.

**Pull request merged**